src/realm/ucx/ucp_internal.cc

```diff
@@ -880,6 +895,33 @@ namespace UCP {
     return true;
   }
 
+  void UCPInternal::add_remote_ep(NodeID peer, const void* blob, size_t bytes) {
+    /* blob layout:  [1-byte dev_index | raw ucp_worker_address] */
+    if(bytes < 2) return;                      // malformed
+
+    const uint8_t      *p     = static_cast<const uint8_t*>(blob);
+    int           rdev  = *p++;          // remote device index
+    const ucp_address_t*addr  = reinterpret_cast<const ucp_address_t*>(p);
+    const size_t        alen  = bytes - 1;     // address length (may be unused)
+
+    rdev = -1; // TODO: FIX ME
+
+    (void)alen;                                // silences “unused” in release
+
+    /* Choose host context, priority 0 – identical to create_eps() */
+    const UCPContext *ctx = get_context_host();
+    UCPWorker        *w   = get_tx_worker(ctx, /*priority=*/0);
```

Comment:
Please do not forget to fix the `priority`, each context has multiple priorities, and each priority maps to a worker.